mount/temp.go

```diff
@@ -0,0 +1,50 @@
+package mount
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+var tempMountLocation = os.TempDir()
```

Comment:
(no response to my previous question, so asking again)

What do you think about using the context for this instead of a global? It can still default to `os.TempDir()` if it's not set in the context:

```go

type tmpPath struct{}

func ContextWithTempDir(ctx context.Context, path string) (context.Context, error) {
    // mkdir path ...
    return context.WithValue(ctx, tmpPath, path), nil
}

func tmpDir(ctx context.Context) string {
    dir := ctx.Value(tmpPath)
    if dir != nil {
        return dir.(string)
    }
    return os.TempDir()
}
```

Reply:
I think its too complex at this point